comps/retrievers/langchain/vdms/retriever_vdms.py

```diff
@@ -0,0 +1,125 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import time
+
+from langchain_community.embeddings import HuggingFaceBgeEmbeddings, HuggingFaceHubEmbeddings
+from langchain_community.vectorstores.vdms import VDMS, VDMS_Client
+from langchain_huggingface.embeddings import HuggingFaceEndpointEmbeddings
+from vdms_config import (  # , HUGGINGFACEHUB_API_TOKEN, INDEX_SCHEMA, VDMS_URL
+    DISTANCE_STRATEGY,
+    EMBED_MODEL,
+    INDEX_NAME,
+    SEARCH_ENGINE,
+    VDMS_HOST,
+    VDMS_PORT,
+)
+
+from comps import (
+    EmbedDoc,
+    SearchedDoc,
+    SearchedMultimodalDoc,
+    ServiceType,
+    TextDoc,
+    opea_microservices,
+    register_microservice,
+    register_statistics,
+    statistics_dict,
+)
+
+tei_embedding_endpoint = os.getenv("TEI_EMBEDDING_ENDPOINT")
+hf_token = os.getenv("HUGGINGFACEHUB_API_TOKEN")
+use_clip = int(os.getenv("USECLIP"))
+
+if use_clip:
+    import sys
+
+    sys.path.append("../../../embeddings/langchain_multimodal/")
+    from embeddings_clip import vCLIP
+
+# Debugging
+all_variables = dir()
+
+for name in all_variables:
```

Comment:
Hi Sameh, could this be debug only? Take advantage of the DEBUG flag in vdma_config.py

Reply:
I've update this since it's US holiday.